processor/stash/stash.go

```diff
@@ -183,48 +186,65 @@ func (st *HandleT) storeErrorsToObjectStorage(jobs []*jobsdb.JobT) (errorJob []E
 			})
 			continue
 		}
-		path := fmt.Sprintf("%v%v.json.gz", tmpDirPath+localTmpDirName, fmt.Sprintf("%v.%v.%v.%v.%v", time.Now().Unix(), config.GetString("INSTANCE_ID", "1"), fmt.Sprintf("%v-%v", jobs[0].JobID, jobs[len(jobs)-1].JobID), uuid, workspaceID))
-		dumps[workspaceID] = path
+		path := filepath.Join(
+			tmpDirPath,
+			localTmpDirName,
+			fmt.Sprintf(
+				"%v.%v.%v.%v.%v.json.gz",
+				time.Now().Unix(),
+				config.GetString("INSTANCE_ID", "1"),
+				fmt.Sprintf("%v-%v", jobs[0].JobID, jobs[len(jobs)-1].JobID),
+				uuid,
+				workspaceID,
+			),
+		)
+		if err := os.MkdirAll(filepath.Dir(path), os.ModePerm); err != nil {
+			panic(fmt.Errorf("creating gz file %q: mkdir error: %w", path, err))
+		}
+		writer, err := misc.CreateGZ(path)
+		if err != nil {
+			panic(err)
+		}
+		writerMap[workspaceID] = path
 		newline := []byte("\n")
 		lo.ForEach(jobsForWorkspace, func(job *jobsdb.JobT, _ int) {
 			rawJob, err := json.Marshal(job)
 			if err != nil {
 				panic(err)
 			}
-			if _, err := gzWriter.Write(path, append(rawJob, newline...)); err != nil {
+			if _, err := writer.Write(append(rawJob, newline...)); err != nil {
+				_ = writer.Close()
 				panic(err)
 			}
 		})
+		if err := writer.Close(); err != nil {
+			panic(err)
+		}
 	}
 
-	err = gzWriter.Close()
-	if err != nil {
-		panic(err)
-	}
 	defer func() {
-		for _, path := range dumps {
-			os.Remove(path)
+		for _, path := range writerMap {
+			_ = os.Remove(path)
 		}
 	}()
 
-	g, ctx := errgroup.WithContext(ctx)
-	g.SetLimit(config.GetInt("Processor.errorBackupWorkers", 100))
+	g, ctx := kitsync.NewEagerGroup(ctx, config.GetInt("Processor.errorBackupWorkers", 100))
 	var mu sync.Mutex
-	for workspaceID, filePath := range dumps {
-		wrkId := workspaceID
-		path := filePath
-		errFileUploader, err := st.fileuploader.GetFileManager(ctx, wrkId)
+	for workspaceID, path := range writerMap {
+		errFileUploader, err := st.fileuploader.GetFileManager(ctx, workspaceID)
 		if err != nil {
-			st.logger.Errorf("Skipping Storing errors for workspace: %s since no file manager is found", workspaceID)
-			mu.Lock()
-			errorJobs = append(errorJobs, ErrorJob{
-				jobs: jobsPerWorkspace[workspaceID],
-				errorOutput: StoreErrorOutputT{
-					Location: "",
-					Error:    err,
-				},
-			})
-			mu.Unlock()
+			st.logger.Errorn("Skipping Storing errors for workspace since no file manager is found", obskit.WorkspaceID(workspaceID))
```

Comment:
Shall we add the error as well?
```suggestion
			st.logger.Errorn("Skipping Storing errors for workspace since no file manager is found", obskit.WorkspaceID(workspaceID), obskit.Error(err))
```